src/sentry/api/endpoints/project_details.py

```diff
@@ -900,11 +900,9 @@ def dynamic_sampling_audit_log(
         :new_raw_dynamic_sampling: The updated dynamic sampling object
 
         """
-        if old_raw_dynamic_sampling is None or new_raw_dynamic_sampling is None:
-            return
 
-        old_rules = old_raw_dynamic_sampling.get("rules", [])
-        new_rules = new_raw_dynamic_sampling.get("rules", [])
+        if old_raw_dynamic_sampling is None and new_raw_dynamic_sampling is None:
+            return
```

Comment:
Please move this check to line 917 instead so that we don't repeat the condition again 
Could be something like 
```
if old_raw_dynamic_sampling is None:
    if new_raw_dynamic_sampling is not None:
         create_audit_entry_for_event(common_audit_data, "SAMPLING_RULE_ADD")
    return
```